datalad/distribution/get.py

```diff
@@ -0,0 +1,232 @@
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar="PATH",
+            doc="""path/name of the requested dataset component. The component
+            must already be known to the dataset.""",
+            nargs="*",
+            constraints=EnsureStr() | EnsureNone()),
+        source=Parameter(
+            args=("-s", "--source",),
+            metavar="LABEL",
+            doc="""label of the data source to be used to fulfill the request.
+            This can be the name of a dataset :term:`sibling` or another known
+            source""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_get_opts=annex_get_opts)
+
+    # Note: May be use 'git annex find --not --in here' to have a list of all
+    # files to actually get and give kind of a progress in terms of number
+    # files processed ...
+
+    @staticmethod
+    @datasetmethod(name='get')
+    def __call__(
+            path=None,
+            source=None,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None,
+            git_opts=None,
+            annex_opts=None,
+            annex_get_opts=None):
+
+        # check parameters:
+        if path is None:
+            raise InsufficientArgumentsError("insufficient information for "
+                                             "getting: requires at least a "
+                                             "path.")
+        # When called from cmdline `path` will be a list even if
+        # there is only one item.
+        # Make sure we deal with the same when called via python API:
+        if not isinstance(path, list):
+            path = [path]
+
+        # resolve path(s):
+        lgr.info("Resolving paths ...")
+        resolved_paths = [resolve_path(p, dataset) for p in path]
+
+        # resolve base dataset:
+        ds = require_dataset(dataset, check_installed=True,
+                             purpose='getting content')
+        lgr.debug("Resolved dataset: %s" % ds)
+
+        # resolve possible subdatasets:
+        lgr.info("Resolving (sub-)datasets ...")
+        resolved_datasets = dict()
+        for p in resolved_paths:
+            if not lexists(p):
+                # skip early:
+                # Note/TODO: This is to be changed, when implementing implicit
+                # install of subdatasets
+                lgr.warning("{0} not found. Ignored.".format(p))
+                continue
+
+            try:
+                p_ds = \
+                    ds.get_containing_subdataset(p,
+                                                 recursion_limit=recursion_limit)
+            except ValueError as e:
```

Comment:
I would not rely on that - make a dedicated exception, and below -- don't use e.message but use `exc_str` helper function (it will include information about original location of the exception which could be helperful etc)